src/screens/Battle.js

```diff
@@ -56,8 +88,12 @@ const Battle = ({navigation, route}) => {
           <Text style={styles.valueTreasure}>{quest.reward}</Text>
         </View>
         <MainButton
-          title={hasBugs ? 'Lutar' : 'Farmar'}
-          onPress={() => navigation.navigate('StartBattle')}
+          title={buttonLabel}
+          onPress={
+            hasBugs
+              ? () => navigation.navigate('StartBattle', {quest})
+              : wonBattle
+          }
```

Comment:
You could create a function to handle this logic so the JSX is cleaner.

Like
```
const handleMainButtonPress = () => {
  ...Logic highlighted
}
```

And then use it like
`
onPress={handleMainButtonPress}
`